ClassNotFoundException: RustReferenceContributor

**RON Plugin version**: 0.3.4
**IDE version**: RustRover 2024.3.4
**Runtime version**: 21.0.5+8-b631.30 aarch64 (JCEF 122.1.9)
**VM**: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

First of all, thank you for the RON support plugin, it was missing for a long time on RustRover.
There appears to currently be a small crash in the plugin, which sometimes cause the syntaxic support to not work, and can only be fixed by restarting the IDE
Here is the content of one such`stacktrace.txt`
```txt
com.intellij.diagnostic.PluginException: com.github.amondeshir.rustroverjronremix.rust.RustReferenceContributor PluginClassLoader(plugin=PluginDescriptor(name=RON Extended Support for Rust Rover, id=com.github.amondeshir.rustroverronremix, descriptorPath=plugin.xml, path=~/Library/Application Support/JetBrains/RustRover2024.3/plugins/RON Extended Support for Rust Rover, version=0.3.4, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.valgrind, descriptorPath=com.jetbrains.rust.valgrind.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.valgrind, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rider, descriptorPath=com.jetbrains.rust.rider.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.rider, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.profiler, descriptorPath=com.jetbrains.rust.profiler.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.profiler, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebugRustRover, descriptorPath=com.jetbrains.rust.nativeDebugRustRover.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebugRustRover, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebugIdea, descriptorPath=com.jetbrains.rust.nativeDebugIdea.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebugIdea, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebug, descriptorPath=com.jetbrains.rust.nativeDebug.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebug, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.mlCompletion, descriptorPath=com.jetbrains.rust.mlCompletion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ml, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.llm, descriptorPath=com.jetbrains.rust.llm.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.llm, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.js, descriptorPath=com.jetbrains.rust.js.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.js, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.idea, descriptorPath=com.jetbrains.rust.idea.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ide.idea, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.duplicates, descriptorPath=com.jetbrains.rust.duplicates.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ide.clones, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.grazie, descriptorPath=com.jetbrains.rust.grazie.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.grazie, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.debugger, descriptorPath=com.jetbrains.rust.debugger.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.debugger, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.coverage, descriptorPath=com.jetbrains.rust.coverage.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.coverage, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.copyright, descriptorPath=com.jetbrains.rust.copyright.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.copyright, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.clion, descriptorPath=com.jetbrains.rust.clion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.clion, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.appcode, descriptorPath=com.jetbrains.rust.appcode.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.appcode, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.terminal, descriptorPath=com.jetbrains.rust.terminal.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.terminal, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.sql, descriptorPath=com.jetbrains.rust.sql.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.sql, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rustrover-only, descriptorPath=com.jetbrains.rust.rustrover-only.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.rover.only, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.inline.completion, descriptorPath=com.jetbrains.rust.inline.completion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.inline.completion, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.fullLine, descriptorPath=com.jetbrains.rust.fullLine.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.fullLine, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rml-dfa, descriptorPath=com.jetbrains.rust.rml-dfa.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.dfa, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, descriptorPath=plugin.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust, isBundled=true), )
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:952)
	at com.intellij.serviceContainer.LazyExtensionInstance.createInstance(LazyExtensionInstance.java:57)
	at com.intellij.serviceContainer.LazyExtensionInstance.getInstance(LazyExtensionInstance.java:46)
	at com.intellij.serviceContainer.BaseKeyedLazyInstance.getInstance(BaseKeyedLazyInstance.java:38)
	at com.intellij.openapi.util.KeyedExtensionCollector.instantiate(KeyedExtensionCollector.java:200)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensionsFromExtensionPoint(KeyedExtensionCollector.java:162)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensions(KeyedExtensionCollector.java:135)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:131)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:16)
	at com.intellij.openapi.util.KeyedExtensionCollector.forKey(KeyedExtensionCollector.java:112)
	at com.intellij.lang.LanguageExtension.collectAllForLanguage(LanguageExtension.java:122)
	at com.intellij.lang.LanguageExtension.allForLanguage(LanguageExtension.java:115)
	at com.intellij.lang.LanguageExtension.allForLanguageOrAny(LanguageExtension.java:143)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.createRegistrar(ReferenceProvidersRegistryImpl.java:63)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.getRegistrar(ReferenceProvidersRegistryImpl.java:101)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistryImpl.doGetReferencesFromProviders(ReferenceProvidersRegistryImpl.java:122)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.lambda$getReferencesFromProviders$0(ReferenceProvidersRegistry.java:44)
	at com.intellij.psi.util.CachedValuesManager$1.compute(CachedValuesManager.java:173)
	at com.intellij.psi.impl.AbstractPsiCachedValue.doCompute(PsiCachedValueImpl.kt:21)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:299)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:37)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:299)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:113)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:29)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:300)
	at com.intellij.psi.impl.PsiCachedValueImpl$Direct.getValue(PsiCachedValueImpl.kt:121)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:83)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:170)
	at com.intellij.psi.util.CachedValuesManager.getCachedValue(CachedValuesManager.java:136)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:42)
	at com.intellij.psi.impl.source.resolve.reference.ReferenceProvidersRegistry.getReferencesFromProviders(ReferenceProvidersRegistry.java:35)
	at com.github.amondeshir.rustroverronremix.language.psi.RONObjectNameMixin.getReferences(RONObjectNameMixin.kt:16)
	at com.intellij.psi.PsiReferenceServiceImpl.doGetReferences(PsiReferenceServiceImpl.java:39)
	at com.intellij.psi.PsiReferenceServiceImpl.getReferences(PsiReferenceServiceImpl.java:27)
	at com.intellij.codeInsight.highlighting.HyperlinkAnnotator.lambda$static$0(HyperlinkAnnotator.java:83)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.doCompute(PsiParameterizedCachedValue.kt:24)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$3(CachedValueBase.java:299)
	at com.intellij.util.CachedValueBase.computeData(CachedValueBase.java:37)
	at com.intellij.util.CachedValueBase.lambda$getValueWithLock$4(CachedValueBase.java:299)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:113)
	at com.intellij.openapi.util.RecursionGuard.doPreventingRecursion(RecursionGuard.java:29)
	at com.intellij.openapi.util.RecursionManager.doPreventingRecursion(RecursionManager.java:68)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:300)
	at com.intellij.psi.impl.PsiParameterizedCachedValue.getValue(PsiParameterizedCachedValue.kt:18)
	at com.intellij.psi.util.CachedValuesManager.getParameterizedCachedValue(CachedValuesManager.java:97)
	at com.intellij.codeInsight.highlighting.HyperlinkAnnotator.getReferences(HyperlinkAnnotator.java:95)
	at com.intellij.codeInsight.highlighting.HyperlinkAnnotator.annotateContributedReferences(HyperlinkAnnotator.java:65)
	at com.intellij.codeInsight.highlighting.HyperlinkAnnotator.annotate(HyperlinkAnnotator.java:60)
	at com.intellij.codeInsight.daemon.impl.AnnotationHolderImpl.runAnnotatorWithContext(AnnotationHolderImpl.java:218)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.lambda$runAnnotator$2(AnnotatorRunner.java:130)
	at com.intellij.codeInsight.daemon.impl.AnnotationSessionImpl.computeWithSession(AnnotationSessionImpl.java:87)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.runAnnotator(AnnotatorRunner.java:120)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.lambda$runAnnotatorsAsync$0(AnnotatorRunner.java:67)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.tryRunReadAction(AnyThreadWriteThreadingSupport.kt:333)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:972)
	at com.intellij.codeInsight.daemon.impl.AnnotatorRunner.lambda$runAnnotatorsAsync$1(AnnotatorRunner.java:67)
	at com.intellij.concurrency.ApplierCompleter.processArrayItem(ApplierCompleter.java:126)
	at com.intellij.concurrency.ApplierCompleter.processArray(ApplierCompleter.java:207)
	at com.intellij.concurrency.ApplierCompleter.execAll(ApplierCompleter.java:176)
	at com.intellij.concurrency.ApplierCompleter.lambda$exec$0(ApplierCompleter.java:115)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.tryRunReadAction(AnyThreadWriteThreadingSupport.kt:351)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:972)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$2(ApplierCompleter.java:158)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:674)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:749)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:673)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:79)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:169)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapAndRun$1(ApplierCompleter.java:150)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.executeByImpatientReader(AnyThreadWriteThreadingSupport.kt:544)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:176)
	at com.intellij.concurrency.ApplierCompleter.wrapAndRun(ApplierCompleter.java:150)
	at com.intellij.concurrency.ApplierCompleter.exec(ApplierCompleter.java:118)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.ClassNotFoundException: com.github.amondeshir.rustroverjronremix.rust.RustReferenceContributor PluginClassLoader(plugin=PluginDescriptor(name=RON Extended Support for Rust Rover, id=com.github.amondeshir.rustroverronremix, descriptorPath=plugin.xml, path=~/Library/Application Support/JetBrains/RustRover2024.3/plugins/RON Extended Support for Rust Rover, version=0.3.4, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.valgrind, descriptorPath=com.jetbrains.rust.valgrind.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.valgrind, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rider, descriptorPath=com.jetbrains.rust.rider.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.rider, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.profiler, descriptorPath=com.jetbrains.rust.profiler.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.profiler, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebugRustRover, descriptorPath=com.jetbrains.rust.nativeDebugRustRover.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebugRustRover, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebugIdea, descriptorPath=com.jetbrains.rust.nativeDebugIdea.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebugIdea, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.nativeDebug, descriptorPath=com.jetbrains.rust.nativeDebug.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.nativeDebug, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.mlCompletion, descriptorPath=com.jetbrains.rust.mlCompletion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ml, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.llm, descriptorPath=com.jetbrains.rust.llm.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.llm, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.js, descriptorPath=com.jetbrains.rust.js.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.js, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.idea, descriptorPath=com.jetbrains.rust.idea.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ide.idea, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.duplicates, descriptorPath=com.jetbrains.rust.duplicates.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.ide.clones, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.grazie, descriptorPath=com.jetbrains.rust.grazie.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.grazie, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.debugger, descriptorPath=com.jetbrains.rust.debugger.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.debugger, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.coverage, descriptorPath=com.jetbrains.rust.coverage.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.coverage, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.copyright, descriptorPath=com.jetbrains.rust.copyright.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.copyright, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.clion, descriptorPath=com.jetbrains.rust.clion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.clion, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.appcode, descriptorPath=com.jetbrains.rust.appcode.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust.appcode, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.terminal, descriptorPath=com.jetbrains.rust.terminal.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.terminal, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.sql, descriptorPath=com.jetbrains.rust.sql.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.sql, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rustrover-only, descriptorPath=com.jetbrains.rust.rustrover-only.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.rover.only, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.inline.completion, descriptorPath=com.jetbrains.rust.inline.completion.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.inline.completion, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.fullLine, descriptorPath=com.jetbrains.rust.fullLine.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.fullLine, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, moduleName=com.jetbrains.rust.rml-dfa, descriptorPath=com.jetbrains.rust.rml-dfa.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=com.jetbrains.rust.dfa, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.impl, descriptorPath=intellij.platform.vcs.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.log.impl, descriptorPath=intellij.platform.vcs.log.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.vcs.dvcs.impl, descriptorPath=intellij.platform.vcs.dvcs.impl.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=IDEA CORE, id=com.intellij, moduleName=intellij.platform.collaborationTools, descriptorPath=intellij.platform.collaborationTools.xml, path=~/Applications/RustRover.app/Contents/lib, version=243.23654.180, package=null, isBundled=true), PluginDescriptor(name=Rust, id=com.jetbrains.rust, descriptorPath=plugin.xml, path=~/Applications/RustRover.app/Contents/plugins/intellij-rust, version=243.23654.180, package=org.rust, isBundled=true), )
	at com.intellij.serviceContainer.ComponentManagerImplKt.doLoadClass(ComponentManagerImpl.kt:1462)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:944)
	... 80 more
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -4,7 +4,7 @@
 pluginGroup = com.github.amondeshir.rustroverjronremix
 pluginName = RON Extended Support for Rust Rover
 pluginRepositoryUrl = https://github.com/AmonDeShir/rustrover-ron-remix/
-pluginVersion = 0.3.5
+pluginVersion = 0.3.6
 
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 252.0

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,10 +1,10 @@
 # IntelliJ Platform Artifacts Repositories
 # -> https://www.jetbrains.org/intellij/sdk/docs/reference_guide/intellij_artifacts.html
 
-pluginGroup = com.github.amondeshir.rustroverjronremix
+pluginGroup = com.github.amondeshir.rustroverronremix
 pluginName = RON Extended Support for Rust Rover
 pluginRepositoryUrl = https://github.com/AmonDeShir/rustrover-ron-remix/
-pluginVersion = 0.3.6
+pluginVersion = 0.3.5
 
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 252.0

--- a/gradle.properties
+++ b/gradle.properties
@@ -4,15 +4,15 @@
 pluginGroup = com.github.amondeshir.rustroverronremix
 pluginName = RON Extended Support for Rust Rover
 pluginRepositoryUrl = https://github.com/AmonDeShir/rustrover-ron-remix/
-pluginVersion = 0.3.5
+pluginVersion = 0.3.6
 
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
-pluginSinceBuild = 252.0
-pluginUntilBuild = 252.*
+pluginSinceBuild = 253.0
+pluginUntilBuild = 261.*
 
 # IntelliJ Platform Properties -> https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html#configuration-intellij-extension
 platformType = RR
-platformVersion = 2025.2
+platformVersion = 2025.3
 
 # Plugin Dependencies -> https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_dependencies.html
 # Example: platformPlugins = com.intellij.java, com.jetbrains.php:203.4449.22

--- a/src/main/kotlin/com/github/amondeshir/rustroverronremix/rust/Inference.kt
+++ b/src/main/kotlin/com/github/amondeshir/rustroverronremix/rust/Inference.kt
@@ -28,7 +28,6 @@ import org.rust.lang.core.types.rawType
 import org.rust.lang.core.types.ty.Ty as RsType
 import org.rust.lang.core.types.ty.TyAdt as RsTypeAdt
 import org.rust.lang.core.types.ty.TyArray as RsTypeArray
-import org.rust.lang.core.types.ty.TyReference as RsTypeRef
 import org.rust.lang.core.types.ty.TySlice as RsTypeSlice
 import org.rust.lang.core.types.ty.TyTuple as RsTypeTuple
 
@@ -193,11 +192,30 @@ private class InferenceBuilder(
      * And Box types can be deserialized.
      */
     private fun RsType.deref(): RsType = when {
-        this is RsTypeRef -> this.referenced.substitute(this.typeParameterValues).deref()
+        this.isRustTyReference() -> this.unwrapRustTyReference()?.deref() ?: this
         this is RsTypeAdt && this.item == this.item.knownItems.Box -> {
             this.typeArguments.first().substitute(this.typeParameterValues).deref()
         }
         else -> this
+    }
+
+    private fun RsType.isRustTyReference(): Boolean =
+        javaClass.name == "org.rust.lang.core.types.ty.TyReference"
+
+    private fun RsType.unwrapRustTyReference(): RsType? {
+        val referenced = javaClass
+            .methods
+            .firstOrNull { it.name == "getReferenced" && it.parameterCount == 0 }
+            ?.invoke(this) as? RsType
+            ?: return null
+
+        val typeParameterValues = javaClass
+            .methods
+            .firstOrNull { it.name == "getTypeParameterValues" && it.parameterCount == 0 }
+            ?.invoke(this) as? Substitution
+            ?: return referenced
+
+        return referenced.substitute(typeParameterValues)
     }
 
     private fun Iterable<RsType>.deref(): List<RsType> = map { it.deref() }

--- a/src/main/resources/META-INF/com.github.amondeshir.rustroverronremix-rust.xml
+++ b/src/main/resources/META-INF/com.github.amondeshir.rustroverronremix-rust.xml
@@ -1,5 +1,5 @@
 <idea-plugin>
     <extensions defaultExtensionNs="com.intellij">
-        <psi.referenceContributor implementation="com.github.amondeshir.rustroverjronremix.rust.RustReferenceContributor" language="RON"/>
+        <psi.referenceContributor implementation="com.github.amondeshir.rustroverronremix.rust.RustReferenceContributor" language="RON"/>
     </extensions>
 </idea-plugin>

